site/en/docs/extensions/reference/runtime/index.md

```diff
@@ -1,6 +1,59 @@
 ---
 api: runtime
 ---
+The runtime API provides methods to support a number of areas of functionality that your extensions
+can use:
+
+Message passing
+
+: These methods support message passing between tasks within the extension (such as background
+  scripts) and with the native system on which the browser is running. See [Message
+  Passing][message-passing] for an overview of the subject.
+  
+  * [runtime.connect(extensionId: string, connectInfo: object): Port](/docs/extensions/reference/runtime/#method-connect)
+  * [runtime.connectNative(application: string): Port]()
+  * [runtime.sendMessage(extensionId?: string, message: any, options: object, responseCallback: function)]()
+  * [runtime.sendNativeMessage(application: string, message: object, responseCallback: function)]()
```

Comment:
Thinking about this more, I worry that these are just going to get out of date.  Any time we add a new method, change a signature, change how we represent the signatures in the docs, etc, they are invalidated.

What do you think about:
- Keeping the categories
- Having each category reference and link to "such as" methods

Example:

Message passing
: These methods support passing messages between different contexts.  These can be used either for contexts within your own extension (with [runtime.connect()] and [runtime.sendMessage()], or with native applications on the machine leveraging [native messaging](https://developer.chrome.com/docs/apps/nativeMessaging/).  
See [Message  Passing][message-passing] for an overview of the subject.

Reply:
I agree with this concern. I did something similar to your suggestion, but a little more boilerplate-like, so that

- instead of a bullet list of methods (with signatures!), it has a run-in list of method names
- it says each time "Methods in this category include"

WDYT?